app/models/device.rb

```diff
@@ -72,7 +72,82 @@ def pretty_owner
     user&.username || "Anonymous"
   end
 
+  def country
+    @country ||= ISO3166::Country.find_country_by_alpha3(self.location)
+  end
+
   def two_digit_location
-    ISO3166::Country.find_country_by_alpha3(self.location).alpha2
+    country.alpha2
+  end
+
+  def full_country_name
+    country.iso_long_name
+  end
+
+  def ram
+    (self.ram_units * self.ram_capacity_per_unit).to_i
+  end
+
+  def total_memory
```

Comment:
I missed this on my first review, but I feel like this function is named a little inconsistently. It's calculating total disk capacity, but it's called `total_memory` which conventionally would refer to RAM ("storage" would be more clear for this purpose instead).

Also, given that it does something very similar to the `ram` function defined above it, I'd expect there to be some consistency between the names - either `ram` would be renamed to `total_ram`, or this would would be renamed to just `storage`.

Reply:
I've done one better and removed it completely as this method wasn't being used any more.